app/commands/track/retrieve_unimplemented_practice_exercises.rb

```diff
@@ -4,12 +4,19 @@ class Track::RetrieveUnimplementedPracticeExercises
   initialize_with :track
 
   def call
-    prob_specs_repo.active_exercise_slugs -
-      track.practice_exercises.pluck(:slug) -
-      track.git.foregone_exercises
+    prob_specs.active_exercises.reject do |exercise|
+      implemented_exercises.include?(exercise.slug) ||
+        foregone_exercises.include?(exercise.slug)
+    end
   end
 
   private
   memoize
-  def prob_specs_repo = Git::ProblemSpecifications.new
+  def prob_specs = ProblemSpecifications.new
+
+  memoize
+  def implemented_exercises = track.practice_exercises.pluck(:slug).to_set
+
+  memoize
+  def foregone_exercises = track.git.foregone_exercises.to_set
```

Comment:
This is fine if it's called once in a request but will be slow if we ever show a page with all 60 tracks iterating over this. Just to be aware of that.

Reply:
Yeah, true. We could always materialize when we want to